c_common/models/reverse_iptag_multicast_source/reverse_iptag_multicast_source.c

```diff
@@ -981,13 +946,87 @@ bool initialize(uint32_t *timer_period) {
     return true;
 }
 
+void timer_callback(uint unused0, uint unused1) {
+    use(unused0);
+    use(unused1);
+    time++;
+
+    log_debug("timer_callback, final time: %d, current time: %d,"
+              "next packet buffer time: %d", simulation_ticks, time,
+              next_buffer_time);
+
+    if ((infinite_run != TRUE) && (time >= simulation_ticks + 1)) {
+        // close recording channels
+         if (recording_is_channel_enabled(
+                recording_flags, e_recording_channel_spike_history)) {
+            recording_finalise();
+        }
+        log_info("Simulation complete.");
+        log_info("Incorrect keys discarded: %d", incorrect_keys);
+        log_info("Incorrect packets discarded: %d", incorrect_packets);
+
+        // Reset buffering
+        address_t address = data_specification_get_data_address();
+        read_parameters(data_specification_get_region(CONFIGURATION, address));
+        if (buffer_region_size > 0) {
+            setup_buffer_region(data_specification_get_region(
+                BUFFER_REGION, address));
+        }
+        time = UINT32_MAX;
+
+        // Wait for the next run of the simulation
+        spin1_callback_off(TIMER_TICK);
+        event_wait();
+
+        initialise_recording();
+
+        spin1_callback_on(TIMER_TICK, timer_callback, 2);
+
+        return;
+    }
+
+    if (send_packet_reqs &&
+            ((time - last_request_tick) >= TICKS_BETWEEN_REQUESTS)) {
+        send_buffer_request_pkt();
+        last_request_tick = time;
+    }
+
+    if (!msg_from_sdram_in_use) {
+        fetch_and_process_packet();
+    } else if (next_buffer_time < time) {
+        fetch_and_process_packet();
+    } else if (next_buffer_time == time) {
+        eieio_data_parse_packet(msg_from_sdram, msg_from_sdram_length);
+        fetch_and_process_packet();
+    }
+}
+
+void sdp_packet_callback(uint mailbox, uint port) {
```

Comment:
I think this whole part should be pushed further down the api. Essentially, all I have added is attempting (unsuccessfully) to listen for a packet sending the EXIT signal (the sending of which we should also talk about, as I'm not happy with how it's being done). This signal should just set a status bit somewhere that tells the simulation to stop after it's done running or after it wakes up from `event_wait`.